src/lib/math/pcurves/pcurves_impl/pcurves_impl.h

```diff
@@ -458,6 +458,123 @@ class IntMod final {
       */
       constexpr Self invert() const { return pow_vartime(Self::P_MINUS_2); }
 
+      /**
+      * Helper for variable time BEEA
+      *
+      * Note this function assumes that its arguments are in the standard
+      * domain, not the Montgomery domain. invert_vartime converts its argument
+      * out of Montgomery, and then back to Montgomery when returning the result.
+      */
+      static constexpr void _invert_vartime_div2_helper(Self& a, Self& x) {
+         constexpr auto INV_2 = p_div_2_plus_1(Rep::P);
+
+         for(;;) {
+            if((a.m_val[0] & 1) == 1) {
+               break;
+            }
```

Comment:
Couldn't that be the loop bound anyway?

```suggestion
         while((a.m_val[0] & 1) != 1) {
```

Reply:
Yes. I had planned to optimize this so that a is shifted just once, as many times as required, then the loop reducing x runs. But it doesn't seem worth bothering with since I think the average number of iterations is 1 or 2.